### PR TITLE
fix: decode non utf8 if response charset presented

### PR DESCRIPTION
**Pull request by kyungeunni**
ticket: https://ubio-automation.atlassian.net/browse/PD-384

when response.headers['content-type'] includes charset, parse it and decode it with given value.

### CODE REVIEW

File: packages/engine/src/main/util/parse.ts [hunk 1]
```diff
@@ -138,3 +138,13 @@ export function parseEncoding(headers: Headers | { [key: string]: string }) {
 
     return 'utf8';
 }
+
+function getContentType(headers: Headers | { [key: string]: string }) {
+    if (headers instanceof Headers) {
+        return headers.get('content-type');
+    }
+    const entry = Object.entries(headers)
```
Comment (inca) [inline packages/engine/src/main/util/parse.ts:146]: Here's a cooler version of one-liner 🕶️ 

```
return new Map(Object.entries(headers).map(([k, v]) => [k.toLowerCase(), v])).get('content-type') ?? null;
```

**Review by inca — APPROVED** [review]
Looks good, thanks for doing this 👍 

Was kinda surprised that this logic exists in the first place though 🤯

File: packages/engine/src/test/specs/actions/Page.fetch.test.ts [hunk 2]
```diff
@@ -92,6 +92,48 @@ describe('Page.fetch', () => {
...
+                    pipeline: {
+                        pipes: [
+                            {
+                                type: 'Value.getJson',
+                                value: JSON.stringify({
+                                    url: runtime.getUrl('/headers'),
+                                    method: 'GET',
+                                    headers: {
+                                        'content-type': 'text/plain;charset=ISO-8859-1',
+                                        hello: 'SANT VICENÇ DE CALDERS',
```
[the remark this answers is not in the document]
Comment (kyungeunni) [reply packages/engine/src/test/specs/actions/Page.fetch.test.ts:113]: yeah we do have POST echo endpoint - I will add a test using that one, too

File: packages/engine/src/main/util/parse.ts (hunk 1 above)
Comment (kyungeunni) [reply packages/engine/src/main/util/parse.ts:146]: Nice 😎 will update it :D

File: packages/engine/src/test/specs/actions/Page.fetch.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyungeunni) [reply packages/engine/src/test/specs/actions/Page.fetch.test.ts:113]: OK this test is wrong! It doesn't set the response header and also I need to encode the payload with the `ISO-8859-1'` to achieve what I want to test. And I'm a bit struggling with setting the response header with the current test set up.

**Comment by kyungeunni** [comment]
Closing it as it isn't a proper way of fixing the particular use case not working as expected, also it shouldn't change the original way of parsing the data for both browser fetch and node-fetch

**Pull request closed without merging**